### PR TITLE
Fixes #14052: create a rudder server trigger_policy_generation command

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/14052

### CODE REVIEW

File: share/commands/server-trigger-policy-generation [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      DISPLAY_COMMAND=true
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+# check if the API is available
+if netstat -lnt  | grep -q "8080" > /dev/null
```
Comment (amousset) [inline share/commands/server-trigger-policy-generation]: maybe better to use the status API?

Comment (ncharles) [reply share/commands/server-trigger-policy-generation]: ha, i was wondering what was the state of the art for that

File: share/commands/server-trigger-policy-generation [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+if [ ${api_code} -eq 0 ]
+then
+  result=$(filtered_api_call "${API_URL}/api/system/update/policies" "${TOKEN}" "POST" -f)
+  code=$?
+  if [ ${code} -eq 0 ]
+  then
+    echo "Successfully triggered a new policy generation"
+  else
+    echo "Failed to trigger a policy "
```
Comment (amousset) [inline share/commands/server-trigger-policy-generation]: we should return 1 in this case

Comment (amousset) [reply share/commands/server-trigger-policy-generation]: and display the message on stderr

**Comment by ncharles** [comment]
Commit modified

File: share/commands/server-trigger-policy-generation [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# @man +
+# @man *-i*: run the agent in information mode, displays all executed commands
+# @man +
+# @man *-c*: run the agent without color output
+
+. "${BASEDIR}/../lib/common.sh"
+. "${BASEDIR}/../lib/api_call.sh"
+
+TOKEN="$(cat /var/rudder/run/api-token)"
+DISPLAY_COMMAND=false
```
Comment (Fdall) [inline share/commands/server-trigger-policy-generation]: This is never used, it should always be the 5th arg of filtered_api_call, see api_call.sh

File: share/commands/server-trigger-policy-generation [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      DISPLAY_COMMAND=true
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+# check if the API is available
+api_result=$(filtered_api_call "${API_URL}/api/system/status" "${TOKEN}" "GET" -f)
```
Comment (Fdall) [inline share/commands/server-trigger-policy-generation]: This is only available in 5.0+, and the request need an api version, most likely "latest"

File: share/commands/server-trigger-policy-generation [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+api_code=$?
+
+if [ ${api_code} -eq 0 ]
+then
+  result=$(filtered_api_call "${API_URL}/api/system/update/policies" "${TOKEN}" "POST" -f)
```
Comment (Fdall) [inline share/commands/server-trigger-policy-generation]: This is only available in 5.0+, and the request need an api version, most likely "latest"

**Pull request merged**